### PR TITLE
chore(Go): add anagram

### PR DESCRIPTION
**Pull request by Ishantgarg-web**
**PR Checklist:**
- [x] My submission is formatted according to the guidelines in the [contributing guide](https://github.com/MakeContributions/DSA#contribution-guidelines)
- [x] My addition is on refer on the language README.md file
- [x] My addition does not have a spelling problem
- [x] My submission has a proper and user-friendly description of the algorithm
- [x] My submission has the time complexity of the algorithm
- [x] My submission has sample input-output of the program (NOT FOR PYTHON)

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

**Other information:**

### CODE REVIEW

File: algorithms/Java/README.md [hunk 1]
```diff
@@ -74,6 +74,7 @@
 - [Celebrity Problem](stacks/celebrity-problem.java)
```
Comment (ming-tsai) [inline algorithms/Java/README.md:74]: Please remove this file change

File: algorithms/Java/stacks/Next_Greater_Element.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
+/***
+ * Problem Statement:
+ * Given an array arr[] of size N having distinct elements, the task is to find the next greater element for each element of the array in order of their appearance in the array.
```
Comment (ming-tsai) [inline algorithms/Java/stacks/Next_Greater_Element.java:3]: Please remove this file change

**Review by ming-tsai — CHANGES_REQUESTED** [review]

**Comment by Ishantgarg-web** [comment]
I remove all unnecessary changes..
Now, please review it

**Review by ayo-ajayi — APPROVED** [review]
It was nicely done.
Thank you.

**Review by ming-tsai — APPROVED** [review]

**Comment by ayo-ajayi** [comment]
I noticed that the package name is "main". It should be changed to "string" since the anagram.go file is in the "string" package.

**Comment by ming-tsai** [comment]
> I noticed that the package name is "main". It should be changed to "string" since the anagram.go file is in the "string" package.
> 
> Also, the main function should be given another name. But make sure the name is written in PascalCase so that it can be exported for use in another package.
> 
> This is a package that should be called by another program.

@ayo-ajayi, could you make a new pull request for enhancement?

**Comment by ayo-ajayi** [comment]
> 

done

**Pull request merged**